### PR TITLE
Set model func

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
Creates a `setModel` function that only updates a single model in the `models` state array. 

`setModel` takes an index and the change applied to that model

closes issue #94

### CODE REVIEW

File: src/components/Controls/ColorMapControls.jsx [hunk 1]
```diff
@@ -22,16 +22,14 @@ function ColorMapControls({ model, modelIdx, setModels }) {
           aria-labelledby="ColorMap dropdown"
           value={model.colorMap.name}
           onChange={(newColorMapName) =>
-            setModels((models) => [
```
Comment (mcmcgrath13) [inline src/components/Controls/ColorMapControls.jsx:25]: I'd consider making these even more specific to the component, so this prop would be `setColorMap` and then on the parent the function would be something like `setColorMap={(colorMap) => setModel({colorMap}, index)`

The `modelIndex` is really the business of the parent and I don't think the child needs to know about it at all.

Comment (RobertGemmaJr) [reply src/components/Controls/ColorMapControls.jsx:25]: It's causing some weird issues with `usecallback`. When I make a custom `setTransferFunction` it's causing the <TransferFunctionControls />` to re-run on every render

Comment (RobertGemmaJr) [reply src/components/Controls/ColorMapControls.jsx:25]: Seems like I need to make each individual function (`setColorMap`, `setTransferFunction`) a `useCallback` function but `setModel` is already a callback function

Comment (RobertGemmaJr) [reply src/components/Controls/ColorMapControls.jsx:25]: Pulling `setTransferFunction` out of the useEffect dependency array (in `TransferFunctionControls`) seems to have done the trip. eslint is mad but I disabled the warning, is it safe to do so?

File: src/components/Controls/TransferFunctionControls.jsx [hunk 2]
```diff
@@ -122,18 +121,16 @@ function TransferFunctionControls({
...
+    setTransferFunction(
+      canvasPoints.map((p) => {
+        return {
+          x: scaleTransferFunctionToCanvasX.invert(p.x),
+          y: scaleTransferFunctionToCanvasY.invert(p.y),
+        };
+      })
     );
-  }, [canvasPoints, modelIdx, pointHovering, pointDragging, setModel]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (mcmcgrath13) [inline src/components/Controls/TransferFunctionControls.jsx:132]: I think that's probably fine in this case to ignore the `setTransferFunction` prop here - @broarr could you sanity check?

**Review by mcmcgrath13 — APPROVED** [review]
pending @broarr 's sanity check

**Comment by RobertGemmaJr** [comment]
I chatted with Brad in person, he said it's all set! Should definitely start working on moving to context though

**Pull request merged**